### PR TITLE
Adds a parallell gzip library

### PR DESCRIPTION
**Pull request by oleorhagen**
This PR replaces the std Golang gzip library with a library adding parallelism to the compression library, this gives a rather substantial improvement of nearly `3x` when run locally on my laptop.

Some profiles:

### Master
```
File: mender-artifact
Build ID: c687cacd156a1724313d2b22b2377fc73952fdbc
Type: cpu
Time: Apr 16, 2020 at 11:11am (CEST)
Duration: 9.54mins, Total samples = 5.77mins (60.49%)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) top10
Showing nodes accounting for 286.34s, 82.66% of 346.39s total
Dropped 129 nodes (cum <= 34.64s)
Showing top 10 nodes out of 34
      flat  flat%   sum%        cum   cum%
   132.65s 38.29% 38.29%    251.90s 72.72%  compress/flate.(*compressor).deflate
    53.72s 15.51% 53.80%     53.72s 15.51%  crypto/sha256.block
    47.52s 13.72% 67.52%     47.63s 13.75%  compress/flate.(*compressor).findMatch
    42.10s 12.15% 79.68%     42.97s 12.41%  syscall.Syscall
     9.61s  2.77% 82.45%     56.51s 16.31%  compress/flate.(*huffmanBitWriter).indexTokens
     0.33s 0.095% 82.55%     47.35s 13.67%  compress/flate.(*huffmanEncoder).generate
     0.13s 0.038% 82.58%     67.48s 19.48%  compress/flate.(*huffmanBitWriter).writeBlock
     0.13s 0.038% 82.62%    342.06s 98.75%  io.copyBuffer
     0.08s 0.023% 82.64%    268.32s 77.46%  archive/tar.(*regFileWriter).Write
     0.07s  0.02% 82.66%    268.39s 77.48%  archive/tar.(*Writer).Write
```

### With parallelism

```
File: mender-artifact
Build ID: a04c7c2b4bc3adae3a2bb712f40d12715a64a890
Type: cpu
Time: Apr 16, 2020 at 12:07pm (CEST)
Duration: 3.52mins, Total samples = 1.83mins (51.88%)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) top 10
Showing nodes accounting for 89.91s, 81.97% of 109.68s total
Dropped 323 nodes (cum <= 10.97s)
Showing top 10 nodes out of 35
      flat  flat%   sum%        cum   cum%
    52.39s 47.77% 47.77%     52.39s 47.77%  crypto/sha256.block
    37.19s 33.91% 81.67%     37.70s 34.37%  syscall.Syscall
     0.06s 0.055% 81.73%     52.46s 47.83%  io.(*multiWriter).Write
     0.05s 0.046% 81.77%     16.06s 14.64%  archive/tar.(*regFileWriter).Write
     0.05s 0.046% 81.82%     87.20s 79.50%  io.copyBuffer
     0.04s 0.036% 81.86%     18.43s 16.80%  syscall.read
     0.04s 0.036% 81.89%     19.35s 17.64%  syscall.write
     0.03s 0.027% 81.92%     16.09s 14.67%  archive/tar.(*Writer).Write
     0.03s 0.027% 81.95%     18.60s 16.96%  os.(*File).Read
     0.03s 0.027% 81.97%     19.42s 17.71%  os.(*File).Write
```

Where the compression has actually disappeared from the top 10 most expensive functions time-wise!

With the addition of #262 this should in total give a pretty decent performance improvement for end users.

I am little unsure of whether or not to add a `Changelog` here.

-O

### CODE REVIEW

File: LIC_FILES_CHKSUM.sha256 [hunk 1]
```diff
@@ -15,6 +15,7 @@ a284e538f3e5649ae6a8f32da2bbd18e9daabf2c1ebbfb99ed92e797b159be1a  vendor/github.
 # ISC licenses.
 1b93a317849ee09d3d7e4f1d20c2b78ddb230b4becb12d7c224c927b9d470251  vendor/github.com/davecgh/go-spew/LICENSE
 a94710b55e03b5285f77d048c5ba61bb9d6ee04a06c0eb90e68821e11b0c707a  vendor/gopkg.in/yaml.v2/LICENSE.libyaml
+16f848582e4b276a7392cd34496b7a33d6f65c0e190c163ff3a056a7c61219ce  vendor/github.com/klauspost/compress/LICENSE
```
Comment (kacf) [inline LIC_FILES_CHKSUM.sha256:18]: Please add this under BSD 3 Clause licenses.

Comment (oleorhagen) [reply LIC_FILES_CHKSUM.sha256:18]: How did you know it was a BSD 3 license?

Comment (kacf) [reply LIC_FILES_CHKSUM.sha256:18]: Years of experience, my friend!

No, seriously, they're always quoting the same license, so it's pretty easy to recognize it once you've looked at a few. The "3 Clause" part is simply the three bullet points, and I skimmed to see that the text was correct.

Comment (mirzak) [reply LIC_FILES_CHKSUM.sha256:18]: It would be more impressive if you recognized it by the SHA256 :)

Comment (oleorhagen) [reply LIC_FILES_CHKSUM.sha256:18]: hahaha, that would have been an amazing bluff. I might have fallen for that one actually :p

**Pull request merged**